mantidimaging/gui/test/gui_system_spectrum_test.py

```diff
@@ -56,3 +57,71 @@ def test_add_roi(self) -> None:
             self.assertEqual(final_roi_count, initial_roi_count + 1)
             self.assertIn(f'roi_{i}', self.spectrum_window.roi_table_model.roi_names())
             self.assertIn(f'roi_{i}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_remove_roi(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        initial_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        QTest.mouseClick(self.spectrum_window.removeBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        final_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        self.assertEqual(final_roi_count, initial_roi_count - 1)
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.roi_table_model.roi_names())
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_change_roi_color(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+
+        new_color = (255, 0, 0, 255)
+        self.spectrum_window.spectrum_widget.change_roi_colour('roi_1', new_color)
+
+        self.assertEqual(self.spectrum_window.spectrum_widget.roi_dict['roi_1'].colour, new_color)
```

Comment:
This could step back a level, as `change_roi_colour()` is quite a deep place to call for system test.

I think it would work to mock `SpectrumROI.openColorDialog` and then call `onChangeColor`